framework/src/main/java/org/checkerframework/framework/util/typeinference8/constraint/ConstraintSet.java

```diff
@@ -162,26 +184,38 @@ public void remove(ConstraintSet subset) {
    */
   public ConstraintSet getClosedSubset(Dependencies dependencies) {
     ConstraintSet subset = new ConstraintSet();
-    Set<Variable> inputDependencies = new LinkedHashSet<>();
-    Set<Variable> outDependencies = new LinkedHashSet<>();
+    // Collection of all outputs of the constraints in this set.
+    Set<Variable> allOutputs = new LinkedHashSet<>();
+    for (Constraint constraint : list) {
+      if (constraint instanceof TypeConstraint) {
+        allOutputs.addAll(((TypeConstraint) constraint).getOutputVariables());
+      }
+      // No other constraints have output variables
+    }
+
+    // Find a subset of this set where the following is true for all the constraints in the subset:
```

Comment:
This does a greedy search.  Is that an important feature of the algorithm, or simply the implementation that you chose?  Please document either way.